### PR TITLE
Refactor GitHub pr (collect and extract) for raw json(merge #1319 first)

### PR DESCRIPTION
**Pull request by warren830**
### Summary
Refactor github pr collector

### Key Points

- [ ] This is a breaking change
- [ ] New or existing documentation is updated

### Description
Describe what this PR does, and aims to solve in a few sentences.

### Does this close any open issues?
Please mention the issues here.

### Current Behavior
Describe the current behaviour of this issue, if relevant.

### New Behavior
Describe the new behaviour updated in this issue, if relevant.

### Screenshots
raw
image
pr
image

pr labels
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/tasks/github_api_issue_collector.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+type GithubApiParams struct {
+	Owner string
+	Repo  string
+}
+
+var _ core.SubTaskEntryPoint = CollectApiIssues
+
+const BatchSize = 100
+
+type ApiIssuesResponse []IssuesResponse
```
Comment (klesh) [inline plugins/github/tasks/github_api_issue_collector.go:24]: move to extractor

File: plugins/github/tasks/github_api_issue_collector.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+type IssuesResponse struct {
```
Comment (klesh) [inline plugins/github/tasks/github_api_issue_collector.go:26]: move to extractor

File: plugins/github/tasks/github_api_issues_extractor.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			*/
+			Table: RAW_ISSUE_TABLE,
+		},
+		Extract: func(row *helper.RawData) ([]interface{}, error) {
+			body := &ApiIssuesResponse{}
+			err := json.Unmarshal(row.Data, body)
+			if err != nil {
+				return nil, err
+			}
+			// need to extract 3 kinds of entities here
```
Comment (klesh) [inline plugins/github/tasks/github_api_issues_extractor.go:38]: 2 kinds?

File: plugins/github/tasks/github_api_pull_request_collector.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	"github.com/merico-dev/lake/plugins/github/models"
+)
+
+const RAW_PULL_REQUEST_TABLE = "github_api_pull_requests"
+
+// this struct should be moved to `gitub_api_common.go`
+
+var _ core.SubTaskEntryPoint = CollectApiPullRequests
+
+type ApiPullRequestResponse []GithubApiPullRequest
```
Comment (klesh) [inline plugins/github/tasks/github_api_pull_request_collector.go:18]: move to extractor

File: plugins/github/tasks/github_api_issue_collector.go (hunk 1 above)
Comment (warren830) [reply plugins/github/tasks/github_api_issue_collector.go]: done

**Review by klesh — APPROVED** [review]

**Pull request merged**